app/src/main/java/me/hufman/androidautoidrive/music/controllers/SpotifyAppController.kt

```diff
@@ -260,8 +271,30 @@ class SpotifyAppController(context: Context, val remote: SpotifyAppRemote): Musi
 	}
 
 	/**
-	 * Creates the QueueMetadata for the current queue containing the title, subtitle, queue songs,
-	 * and queue cover art.
+	 * Creates the [QueueMetadata] for the current player context using the app remote API.
+	 */
+	private fun createQueueMetadata(playerContext: PlayerContext) {
+		if (queueUri != null && queueItems.isEmpty()) {
+			val listItem = ListItem(queueUri, queueUri, null, playerContext.title, playerContext.subtitle, false, true)
+			loadPaginatedItems(listItem, { queueUri == playerContext.uri }) {
+				queueItems = it
+
+				// shuffle play button somehow gets returned with the rest of the tracks when loading an album
+				if (queueItems.isNotEmpty() && queueItems[0].artist == "" && queueItems[0].title == "Shuffle Play") {
```

Comment:
Instead of editing the main `queueItems` and risking a rare race condition, it might be slightly better to set the queueItems to the `drop`d list:
```
if (queueItems.isNotEmpty() && queueItems[0].artist == "" && queueItems[0].title == "Shuffle Play") {
    queueItems = it.drop(1)
} else {
    queueItems = it
}
```

Reply:
Ah that's a much better approach, updated this code to avoid editing the queueItems copy